task_processing/plugins/mesos/execution_framework.py

```diff
@@ -481,110 +267,95 @@ def create_new_docker_task(self, offer, task_config):
 
         return task
 
-    def stop(self):
-        pass
-
     ####################################################################
     #                   Mesos driver hooks go here                     #
     ####################################################################
 
     def slaveLost(self, drive, slaveId):
-        log.error("Slave lost: {id}".format(id=str(slaveId)))
+        log.warning("Slave lost: {id}".format(id=str(slaveId)))
 
     def registered(self, driver, frameworkId, masterInfo):
-        self.framework_id = frameworkId.value
-        paasta_print("Registered with framework ID %s" % frameworkId.value)
+        log.info("Registered with framework ID {id}".format(id=frameworkId.value))
 
-        self.reconcile_start_time = time.time()
-        driver.reconcileTasks([])
+    def reregistered(self, driver, frameworkId, masterInfo):
+        log.warning("Registered with framework ID {id}".format(id=frameworkId.value))
 
     def resourceOffers(self, driver, offers):
-        if self.frozen:
-            return
+        if self.driver is None:
```

Comment:
i understand that in current implementation we need driver to be accessible from inside the framework, but i'm worried this will lead to bugs when driver object becomes stale? not sure how to go about solving this. reassign self.driver every time mesos calls framework methods?